aten/src/ATen/TensorOptions.h

```diff
@@ -17,10 +17,29 @@ namespace at {
 
 /// A class to encapsulate construction axes of a `Tensor`.
 /// `TensorOptions` is a virtual class to enable overriding of certain methods
-/// by subclasses in other libraries, such as PyTorch. In PyTorch, there is a
-/// `torch::TensorOptions` subclass of this `TensorOptions`, which changes
-/// `type()` to return a variable type instead of a tensor type, such that
-/// variables are created inside factory methods, instead of tensors.
+/// by subclasses in other libraries, such as PyTorch.
+///
+/// WARNING: In PyTorch, there are torch:: variants of factory functions,
+/// e.g., torch::zeros for at::zeros.  These return Variables (while the
+/// stock ATen functions return plain Tensors).  If you mix these functions
+/// up, you WILL BE SAD.
+///
+/// Rather than use the constructor of this class directly, you should prefer to
+/// use the constructor functions, and then chain setter methods on top of them.
```

Comment:
'chain setter methods' is a confusing term to me, but maybe it's common?

Reply:
Method chaining is the standard term for what is going on here: https://en.wikipedia.org/wiki/Method_chaining